docs/data/data-grid/group-pivot/RowGroupingGetGroupingCriteriaRows.js

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  gridColumnVisibilityModelSelector,
+  GridEvents,
+  useGridApiRef,
+} from '@mui/x-data-grid-pro';
+import { useMovieData } from '@mui/x-data-grid-generator';
+import Alert from '@mui/material/Alert';
+
+const INITIAL_GROUPING_COLUMN_MODEL = ['company', 'director'];
+
+const useKeepGroupingColumnsHidden = (apiRef, columns, initialModel, leafField) => {
```

Comment:
https://github.com/mui/mui-x/pull/4319

I don't think the pain is the same for us and the developer
For us it is the fact that we have dozens of demos and we don't want to copy the same code dozens of time.
Users can do what we do in #4319 and copy it in a single place of there codebase.

But if we want to support it with first party code, I think it would be interesting to see if we want a prop or a utility hook the user can import and instantiate themselves.

____

> GridEvents.rowGroupingModelChange in useGridColumns

It would mean having the community plan aware of the premium features. If we do want to customize the behavior with a prop, it should probably by in `useGridGroupingColumns`.

Reply:
My experience, when I tried to make demo for the blog article is the following:

- I looked at the doc and activated the experimental feature
- played a bit and quickly realized that having duplicate data in a row seems strange
- I checked in doc demos if the behavior was the same, and found that it was not the case for all, so I opened the code and copy past this big hook

From this experience, I think that developers' pain is to find information about how to handle visibility when grouping.

So it can be set in a package but should be documented

And it should be in `x-data-grid-premium`, otherwise, as a dev, I would think "ok, I need this hook, let's import it" and then get the error "useKeepGroupingColumnsHidden is not exported by ..."

Reply:
> So it can be set in a package but should be documented

Adding a small paragraph in the doc "Hiding the columns used for grouping" could easily be done.

Can we continue the discussion in #4319 ?